### ISSUE TITLE
[4.0.0-beta.3 to beta.7] RTL mode: margin-start/end & padding-start/end do not work

### ISSUE THREAD

**Issue by abennouna** [ionic-team/ionic#15341]
# Bug Report

**Ionic Info**
```
Ionic:

   ionic (Ionic CLI)          : 4.1.1 (~/.nvm/versions/node/v8.11.1/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.3
   @angular-devkit/core       : 0.7.5
   @angular-devkit/schematics : 0.7.5
   @angular/cli               : 6.1.5
   @ionic/ng-toolkit          : 1.0.7
   @ionic/schematics-angular  : 1.0.5

System:

   NodeJS : v8.11.1 (~/.nvm/versions/node/v8.11.1/bin/node)
   npm    : 5.6.0
   OS     : macOS High Sierra
```

**Describe the Bug**
The 4 attributes ```margin-start```, ```margin-end```, ```padding-start```, and ```padding-end``` do not work correctly in RTL mode: they render like in LTR mode.

**Steps to Reproduce**
1. Add ```dir="rtl"``` to the HTML root element
2. Use any of the 4 attributes on a div for instance
3. Notice that they render as in LTR mode (demo provided below)

**Related Code**
- Ionic app: https://github.com/abennouna/ionic4-rtl-issue
- StackBlitz: https://stackblitz.com/edit/github-bpgvd6

**Expected Behavior**
The 4 attributes must render as right-to-left mode.

**Additional Context**
I tried with and without ```$app-direction: multi``` but I haven't noticed any change.

**Comment by abennouna** [comment]
After digging a bit, it seems that by default only LTR directives are included.

I am now able to correctly render those attributes by including the relevant scss files instead of the preprocessed css files:

**[globals.scss]**
```css
$app-direction: multi;
@import "~@ionic/angular/dist/css/ionic";
@import "~@ionic/angular/dist/themes/ionic.globals.md";
```

**[index.html]**
```html
<html lang="en" dir="ltr">
...
```
So it seems a bit like for Ionic 3.

Is this the correct approach? I updated the repo on https://github.com/abennouna/ionic4-rtl-issue

**Comment by abennouna** [comment]
With beta.6:
```css
@import '~@ionic/angular/dist/css/normalize';
@import '~@ionic/angular/dist/css/core';
@import '~@ionic/angular/dist/css/structure';
@import '~@ionic/angular/dist/css/typography';
@import '~@ionic/angular/dist/css/utils.bundle';
@import '~@ionic/angular/dist/themes/ionic.globals.md';
```

**Comment by paulstelzer** [comment]
Is this issue still exist in beta.17?

**Comment by abennouna** [comment]
Yes: RTL mode is broken out of the box, and it's not clear how we can make it work with static CSS files.

As a workaround, we [use SCSS files for the moment](https://github.com/ionic-team/ionic/issues/15341#issuecomment-417674634).

You can check the [updated demo here](https://abennouna.github.io) and the [demo repository here](https://github.com/abennouna/ionic4-rtl-issue)

**Comment by abennouna** [comment]
@paulstelzer I'l like to try to fix some RTL issues, like this one.

Do you know how the CSS compilation work in v4?

I've been banging my head for some time now, and I can't understand how the RTL styles are supposed to be compiled in the final CSS files.

For instance, the ruleset for `[padding-end]` contains [this line]
(https://github.com/ionic-team/ionic/blob/master/core/src/css/padding.scss#L52):
```css
  --padding-end: #{$padding};
```

As mentioned before, when I import the static CSS file alone, I don't get anything related to RTL styles.

But, even if I change that line with:
```css
  @include ltr() {
    --padding-end: #{$padding};
  }
  @include rtl() {
    --padding-start: #{$padding};
  }
```

and compile the project, I see the CSS does not change at all.

Any help or pointer will be appreciated, thanks!

**Comment by paulstelzer** [comment]
Maybe @brandyscarney or @manucorporat can help you :)

**Comment by mohamedsharaf** [comment]
hope you update demo with sample of all error in rtl and make this as guide like floating input not working in RC0

```
 <ion-item no-padding>
                <ion-label position="floating" item-start>
                    email
                </ion-label>
                <ion-input color="secondary" type="email" name="email"></ion-input>
   </ion-item>
```

**Comment by abennouna** [comment]
Is fixed by #16987.
